straxen/plugins/peak_processing.py

```diff
@@ -331,98 +336,393 @@ def find_n_competing(peaks, windows, fraction):
+    provides = 'peak_ambience'
+    data_kind = 'peaks'
+    save_when = strax.SaveWhen.EXPLICIT
+
+    ambience_time_window_backward = straxen.URLConfig(
+        default=int(2e6),
+        type=int,
+        track=True,
+        help='Search for ambience in this time window [ns]'
+    )
+
+    ambience_divide_t = straxen.URLConfig(
+        default=False,
+        type=bool,
+        track=True,
+        help='Whether to divide area by time'
+    )
+
+    ambience_divide_r = straxen.URLConfig(
+        default=False,
+        type=bool,
+        track=True,
+        help='Whether to divide area by radius'
+    )
+
+    ambient_radius = straxen.URLConfig(
+        default=6.7,
+        type=float,
+        track=True,
+        help='Search for ambience in this radius [cm]'
+    )
+
+    ambience_area_parameters = straxen.URLConfig(
+        default=(5, 60, 60), 
+        type=(list, tuple), 
+        track=True,
+        help='The upper limit of S0, S1, S2 area to be counted'
+    )
+
+    def get_window_size(self):
+        return 10 * self.config['ambience_time_window_backward']
+
+    @property
+    def origin_dtype(self):
+        return ['lh_before', 's0_before', 's1_before', 's2_before', 's2_near']
+
+    def infer_dtype(self):
+        dtype = []
+        for ambience in self.origin_dtype:
+            dtype += [((f"Number of small {' '.join(ambience.split('_'))} a peak", 
+                        f'n_{ambience}'), np.int16), 
+                      ((f"Area sum of small {' '.join(ambience.split('_'))} a peak", 
+                        f's_{ambience}'), np.float32)]
+        dtype += strax.time_fields
+        return dtype
+
+    def compute(self, lone_hits, peaks):
+        return self.compute_ambience(lone_hits, peaks, peaks)
+
+    def compute_ambience(self, lone_hits, peaks, current_peak):
+        # 1. Initialization
+        result = np.zeros(len(current_peak), self.dtype)
+        num_array = np.zeros(len(current_peak), dtype=np.int16)
+        sum_array = np.zeros(len(current_peak), dtype=np.float32)
+
+        # 2. Define time window for each peak, we will find small peaks & lone hits within these time windows
+        roi = np.zeros(len(current_peak), dtype=strax.time_fields)
+        roi['time'] = current_peak['center_time'] - self.config['ambience_time_window_backward']
+        roi['endtime'] = current_peak['center_time']
+
+        # 3. Calculate number and area sum of lonehits before a peak
+        touching_windows = strax.touching_windows(lone_hits, roi)
+        # Calculating ambience
+        self.lonehits_ambience(current_peak, 
+                               lone_hits, 
+                               touching_windows, 
+                               num_array, 
+                               sum_array, 
+                               self.config['ambience_divide_t'])
+        result['n_lh_before'] = num_array
+        result['s_lh_before'] = sum_array
```

Comment:
Can you not pass also directly 
```
result['n_lh_before'] ,
result['s_lh_before'] ,
```
instead of 
```
num_array, 
sum_array, 
```
to your function? Then you do not have to define them and copy the values over.